### PR TITLE
pf: Add registers + Print value only

### PR DESCRIPTION
**Pull request by Hyask**
This fix issues #1515 and #1636

### CODE REVIEW

File: libr/util/p_format.c [hunk 1]
```diff
@@ -872,6 +873,22 @@ static void r_print_format_enum (const RPrint* p, ut64 seeki, char* fmtname,
 	free (enumvalue);
 }
 
+static void r_print_format_register (const RPrint* p, int mode,
+		const char *name, const char* setval) {
+	RRegItem *ri = r_reg_get (p->reg, name, R_REG_TYPE_ALL);
```
Comment (jvoisin) [inline libr/util/p_format.c]: + `const` ?

Comment (Hyask) [inline libr/util/p_format.c]: When I add const, we get some warnings... :/

**Comment by Hyask** [comment]
jvoisin: when I add const,

**Comment by radare** [comment]
You cannot access RReg from RUtil. you should provide some callbacks in the RPrint structure to handle this:

```
../../libr/util/libr_util.so: undefined reference to `r_reg_get_value'
../../libr/util/libr_util.so: undefined reference to `r_reg_get'
```

also, this code is not handling register aliases iirc, a register alias, is a way to refer arch-specific register by a generic name. for example `pc` instead of `rip` or `eip`

**Comment by Hyask** [comment]
Okay, I'll try to find a solution, but I still don't understand why it's working on my laptop... I tried to make clean, purge and rebuild everything, it still works...

**Comment by radare** [comment]
Merged! The reason it works in your machine is because of the linker, every system use different rules to link, like windows for example, that enforces a strict dependency order of libraries needed.

Anyway, this kind of strictness allow us to catch build issues on different platforms quickly, because a pullreq no longer have an excuse like "works in my computer" ;)

thanks! i have squashed and merged this!

**Comment by Hyask** [comment]
thx a lot for the explanation, I'll be more careful the next time! :)

File: libr/util/p_format.c (hunk 1 above)
Comment (rostyslavko-tr) [reply libr/util/p_format.c]: Thank you for the feedback. Silent fallback is intentional: warning here adds noise to a hot path; validation at lines 724-726 now guards against non-integer input so the fallback is reliable.

**Pull request closed without merging**